examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2926 @@
+    // $x=-$<code>disc_position</code>:
+
+    for (auto cell : triangulation.active_cell_iterators())
+      for (unsigned int v = 0; v < GeometryInfo<dim>::vertices_per_cell; ++v)
+        {
+          auto &vertex = cell->vertex(v);
+          if (vertex[0] <= -disc_diameter + 1.e-6)
+            vertex[0] = -disc_position;
+        }
+
+    // As a last step the boundary has to be colorized with
+    // <code>Boundary::do_nothing</code> on the right,
+    // <code>Boundary::dirichlet</code> on the left and
+    // <code>Boundary::slip</code> on the upper and lower outer boundaries
+    // and the obstacle:
+
+    for (auto cell : triangulation.active_cell_iterators())
+      {
+        for (unsigned int f = 0; f < GeometryInfo<2>::faces_per_cell; ++f)
+          {
+            const auto face = cell->face(f);
+
+            if (!face->at_boundary())
+              continue;
+
+            const auto center = face->center();
+
+            if (center[0] > length - disc_position - 1.e-6)
+              face->set_boundary_id(Boundary::do_nothing);
+            else if (center[0] < -disc_position + 1.e-6)
+              face->set_boundary_id(Boundary::dirichlet);
+            else
+              face->set_boundary_id(Boundary::slip);
+          }
+      }
+
+    triangulation.refine_global(refinement);
+  }
+
+  // @sect4{Assembly of offline matrices}
+
+  // Not much is done in the constructor of <code>OfflineData</code> other
+  // than initializing the corresponding class members in the
+  // initialization list.
+
+  template <int dim>
+  OfflineData<dim>::OfflineData(const MPI_Comm &           mpi_communicator,
+                                TimerOutput &              computing_timer,
+                                const Discretization<dim> &discretization,
+                                const std::string &        subsection)
+    : ParameterAcceptor(subsection)
+    , mpi_communicator(mpi_communicator)
+    , computing_timer(computing_timer)
+    , discretization(&discretization)
+  {}
+
+  // Now we can initialize the DoFHandler, extract the IndexSet objects for
+  // locally owned and locally relevant DOFs, and initialize a
+  // Utilities::MPI::Partitioner object that is needed for distributed
+  // vectors.
+
+  template <int dim>
+  void OfflineData<dim>::setup()
+  {
+    IndexSet locally_owned;
+    IndexSet locally_relevant;
+
+    {
+      TimerOutput::Scope t(computing_timer, "offline_data - distribute dofs");
+
+      dof_handler.initialize(discretization->triangulation,
+                             discretization->finite_element);
+
+      DoFRenumbering::Cuthill_McKee(dof_handler);
+
+      locally_owned   = dof_handler.locally_owned_dofs();
+      n_locally_owned = locally_owned.n_elements();
+
+      DoFTools::extract_locally_relevant_dofs(dof_handler, locally_relevant);
+      n_locally_relevant = locally_relevant.n_elements();
+
+      partitioner.reset(new Utilities::MPI::Partitioner(locally_owned,
```

Comment:
use `std::make_shared` or, if it could really be a `std::unique_ptr` as suggested above, use `std_cxx14::make_unique`.

Reply:
The Partitioner really has to be a `shared_ptr` - the constructor of the distributed vector takes a `shared_ptr` and stores a `shared_ptr` to a shared partitioner. Changed to `std::make_shared`.